src/aggregationserver/databaseclient.h

```diff
@@ -92,6 +92,16 @@ class DatabaseClient {
         const std::vector<std::string>& condition_values
     );
 
+    void UpdateShutdownTime(
+        int experiment_run_id,
+        std::string end_time
```

Comment:
const string refs here and in UpdateLastSampleTime would be a good idea.

Reply:
Addresed in [this commit](https://github.com/cdit-ma/logan/pull/40/commits/1334d762696ed12fafe4e5cf65bbba7f98218b84)